domain/cert/certificate.go

```diff
@@ -0,0 +1,82 @@
+package cert
+
+import (
+	"errors"
+	"time"
+
+	"github.com/satori/go.uuid"
+)
+
+// NewCertificate creates a Certificate and ensures that the activity properly set empty
```

Comment:
Isn't there a way in go to define private functions? If so, why don't split the constructor into a function `createCertificate()` as well as `assertActivityNotEmpty()` ?

Reply:
Everything constant, method, function or variable that starts with lowercase is not exported but package private.
Everything that starts with upper case is exported and can be used from outside the package.